src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -878,11 +881,20 @@ def median_of_rolling_median_offset(
         else:
             shifted[i, :] = offset
     # Ignore warning for rows with all null values
-    with warnings.catch_warnings():
+    with warnings.catch_warnings(), tempfile.TemporaryDirectory() as tmp_dir:
         warnings.filterwarnings(
             "ignore", category=RuntimeWarning, message="All-NaN slice encountered"
         )
-        return np.nanmedian(shifted, axis=0)
+        array_file = Path(tmp_dir) / "window.npy"
+        np.save(array_file, shifted)
+        del shifted
+        with da.config.set(scheduler="threads", num_workers=2):
+            return da.array.nanmedian(
+                da.array.from_array(
+                    np.load(array_file, mmap_mode="r"), chunks=(21, 10791, 19)
+                ),
+                axis=0,
+            ).compute()
```

Comment:
I wonder if this could also be done in a memory efficient way with DuckDB, which is apparently happy to [query numpy arrays](https://duckdb.org/docs/stable/guides/python/import_numpy)?

Reply:
Wow duckdb can do everything. Unfortunately, I spent some time investigating this option this afternoon and ran into the same issues as polars where you need to offload the array onto disk to get any efficiency gains, and that ends up being a massive bottleneck. Also duckdb can't do `SELECT median(*) FROM table`, so you need to auto-generate a query to compute the median on each individual column and there's 10s of thousands of columns.